app/assets/v2/js/cart-ethereum-polygon.js

```diff
@@ -0,0 +1,430 @@
+const bulkCheckoutAddressPolygon = '0x3E2849E2A489C8fE47F52847c42aF2E8A82B9973';
+
+function objectMap(object, mapFn) {
+  return Object.keys(object).reduce(function(result, key) {
+    result[key] = mapFn(object[key]);
+    return result;
+  }, {});
+}
+
+Vue.component('grantsCartEthereumPolygon', {
+  props: {
+    currentTokens: { type: Array, required: true }, // Array of available tokens for the selected web3 network
+    donationInputs: { type: Array, required: true }, // donationInputs computed property from cart.js
+    grantsByTenant: { type: Array, required: true }, // Array of grants in cart
+    maxCartItems: { type: Number, required: true }, // max number of items in cart
+    grantsUnderMinimalContribution: { type: Array, required: true } // Array of grants under min contribution
+  },
+
+  data: function() {
+    return {
+      network: 'testnet',
+      polygon: {
+        showModal: false, // true to show modal to user, false to hide
+        checkoutStatus: 'not-started', // options are 'not-started', 'pending', and 'complete'
+        estimatedGasCost: '700000'
+      },
+
+      cart: {
+        tokenList: [], // array of tokens in the cart
+        unsupportedTokens: [] // tokens in cart which are not supported by Polygon
+      },
+
+      user: {
+        requiredAmounts: null
+      }
+    };
+  },
+
+  mounted() {
+    window.addEventListener('beforeunload', (e) => {
+      if (this.polygon.checkoutStatus === 'pending') {
+        e.returnValue = 'Polygon checkout in progress. Are you sure you want to leave?';
+      }
+    });
+
+    // Update Polygon checkout connection, state, and data frontend needs when wallet connection changes
+    window.addEventListener('dataWalletReady', async(e) => {
+      await this.onChangeHandler(this.donationInputs);
+    });
+  },
+
+  computed: {
+    /**
+     * @dev List of tokens supported by Polygon + Gitcoin. To add a token to this list:
+     *   1. Make sure the token is top 10 used tokens based on Gitcoin's historical data
+     *   2. Confirm the token exists on Polygon's list of supported tokens: https://mapper.matic.today/
+     *   2. Add the token symbol to the appropriate list below
+     * @dev We hardcode the list from Gitcoin's historical data based on the top ten tokens
+     *   on ethereum chain and also Polygon network used by users to checkout
+     */
+    supportedTokens() {
+      const mainnetTokens = [ 'DAI', 'ETH', 'USDT', 'USDC', 'PAN', 'BNB', 'UNI', 'CELO', 'MASK', 'MATIC' ];
+      const testnetTokens = [ 'MATIC', 'ETH', 'DAI' ];
```

Comment:
^ do we wanna fetch these from DB ? we can use https://gitcoin.co/_administrationperftools/staticjsonenv/

Reply:
isn't that an extra http request? (--performance) => we won't have to change these in a long time